index.js

```diff
@@ -1,59 +1,160 @@
-  log('🔌 Using', provider === 'IPFS_HTTP_API' ? 'local ipfs daemon via http api' : 'js-ipfs node')
-  if (input.length > 1) {
-    log(`🔍 Finding DNSLinks for ${input.length} domains`)
-  }
-  const results = []
-  for (const domain of input) {
-    const spinner = getSpinner(domain)
-    try {
-      const address = await ipfs.dns(domain)
-      const cid = address.slice(6) // trim off the /ipfs/ prefix... TODO: js-ipfs should deal.
-      const stat = await ipfs.object.stat(cid)
-      results.push({ domain, cid, stat })
-      spinner.stop()
-      log('🔗', domain.padEnd(longestDomain), cid, prettyBytes(stat.CumulativeSize))
-    } catch (error) {
-      if (error.message === 'could not resolve name' || error.code === 'ENODATA') {
-        log('⚠️ ', domain.padEnd(longestDomain), 'has no DNSLink TXT record')
-      } else if (error.message === 'not a valid domain name' || error.code === 'ENOTFOUND') {
-        log('⚠️ ', domain.padEnd(longestDomain), 'is not a valid domain')
-      } else {
-        log('⚠️ ', domain.padEnd(longestDomain), error.message || error)
-      }
+'use strict'
+
+function getTimestamp () {
+  return new Date().toISOString()
+    .replace(/:/g, '')
+    .replace('T', '_')
+    .split('.')[0]
+}
+
+function getOptions (opts) {
+  opts = opts || {}
+  opts = Object.assign({
+    lazy: true
+  }, opts)
+  return opts
+}
+
+async function fileExists (ipfs, path) {
+  try {
+    await ipfs.files.stat(path)
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+async function isLazilyCohosted (ipfs, domain) {
+  return fileExists(ipfs, `/cohosting/lazy/${domain}`)
+}
+
+async function isFullyCohosted (ipfs, domain) {
+  return fileExists(ipfs, `/cohosting/full/${domain}`)
+}
+
+async function add (ipfs, domain, opts) {
+  opts = getOptions(opts)
+
+  const cid = await ipfs.resolve(`/ipns/${domain}`)
+  const path = `/cohosting/${opts.lazy ? 'lazy' : 'full'}/${domain}`
+
+  // Create with parents if it is the first time
+  await ipfs.files.mkdir(path, { parents: true })
+  const dirs = await ipfs.files.ls(path)
+  const latest = dirs.map(file => file.name).sort().pop()
+
+  if (latest) {
+    const latestPath = `${path}/${latest}`
+    const stat = await ipfs.files.stat(latestPath)
+
+    if (`/ipfs/${stat.hash}` === cid) {
+      const newPath = `${path}/${getTimestamp()}`
+      await ipfs.files.mv([latestPath, newPath])
+      return ipfs.files.stat(newPath)
     }
   }
-  const totalBytes = results.map(x => x.stat.CumulativeSize).reduce((a, b) => a + b, 0)
-  log(`📦 Total size`, prettyBytes(totalBytes), `for ${results.length} domains`)
 
-  if (flags.pin) {
-    for (const res of results) {
-      const spinner = getSpinner(`Pinning ${res.domain}`)
-      await ipfs.pin.add(res.cid)
-      spinner.stop()
-      log(`📍 Pinned ${res.domain}`)
+  const newPath = `${path}/${getTimestamp()}`
+  await ipfs.files.cp([cid, newPath])
```

Comment:
I'm not sure I understand how lazy works? If `cid` is the CID of a directory (which presumably it is for a website) then `ipfs.files.cp` will copy the directory **and** it's contents right? How does it know when I've visited pages on a website, when are they added to MFS?

Reply:
Gotcha - did not realise nodes were not recursively fetched.

Reply:
N.b. you don't need to wrap the `ipfs.files.cp` args in an array, instead do `await ipfs.files.cp(cid, newPath)`